p/riscv_test.h

```diff
@@ -73,9 +77,13 @@
 #define DELEGATE_NO_TRAPS                                               \
   la t0, 1f;                                                            \
   csrw mtvec, t0;                                                       \
-  csrwi medeleg, 0;                                                     \
-  csrwi mideleg, 0;                                                     \
-  csrwi mie, 0;                                                         \
+  csrr t0, misa;    /* Check if S-mode is implemented ... */            \
+  srli t0, t0, 18;                                                      \
+  andi t0, t0, 1;                                                       \
+  beqz t0, 2f;      /* ... Skip setting medeleg and mideleg if it   */  \
+  csrwi medeleg, 0; /* isn't, since they shouldn't exist without S- */  \
+  csrwi mideleg, 0; /* mode (or the N extension); see vol II pg 26  */  \
+2: csrwi mie, 0;                                                        \
```

Comment:
Similar to the above, this check is redundant.  If medeleg / mideleg don't exist, they'll trap, but mtvec has been set to the following instruction, so execution will continue as normal.